fv3core/stencils/ytp_v.py

```diff
@@ -1,230 +1,166 @@
 
+    return flux
 
-@gtstencil()
-def br_bl_main(q: sd, al: sd, bl: sd, br: sd):
-    with computation(PARALLEL), interval(...):
-        # TODO: add [0, 0, 0] when gt4py bug is fixed
-        bl = get_bl(al=al, q=q)  # noqa
-        br = get_br(al=al, q=q)  # noqa
 
+def _compute_stencil(
+    courant: FloatField,
+    v: FloatField,
+    flux: FloatField,
+    dy: FloatField,
+    dya: FloatField,
+    rdy: FloatField,
+):
+    from __externals__ import i_end, i_start, j_end, j_start, jord, namelist
 
-@gtstencil()
-def br_bl_corner(br: sd, bl: sd):
     with computation(PARALLEL), interval(...):
-        bl = 0
-        br = 0
 
+        if __INLINED(jord < 8):
+            al = yppm.compute_al(v, dy)
 
-def zero_br_bl_corners_south(br, bl):
-    grid = spec.grid
-    corner_domain = (1, 2, grid.npz)
-    if grid.sw_corner:
-        br_bl_corner(br, bl, origin=(grid.is_, grid.js - 1, 0), domain=corner_domain)
-    if grid.se_corner:
-        br_bl_corner(br, bl, origin=(grid.ie + 1, grid.js - 1, 0), domain=corner_domain)
-
-
-def zero_br_bl_corners_north(br, bl):
-    grid = spec.grid
-    corner_domain = (1, 2, grid.npz)
-    if grid.nw_corner:
-        br_bl_corner(br, bl, origin=(grid.is_, grid.je, 0), domain=corner_domain)
-    if grid.ne_corner:
-        br_bl_corner(br, bl, origin=(grid.ie + 1, grid.je, 0), domain=corner_domain)
+            bl = al[0, 0, 0] - v[0, 0, 0]
+            br = al[0, 1, 0] - v[0, 0, 0]
 
+            # Zero corners
+            with horizontal(
+                region[i_start, j_start - 1 : j_start + 1],
+                region[i_start, j_end : j_end + 2],
+                region[i_end + 1, j_start - 1 : j_start + 1],
+                region[i_end + 1, j_end : j_end + 2],
+            ):
+                bl = 0.0
+                br = 0.0
 
-def compute(c, u, v, flux):
+        else:
+            dm = yppm.dm_jord8plus(v)
+            al = yppm.al_jord8plus(v, dm)
+
+            assert __INLINED(jord == 8)
+            # {
+            bl, br = yppm.blbr_jord8(v, al, dm)
+            # }
+
+            assert __INLINED(namelist.grid_type < 3)
+            # {
+            with horizontal(region[:, j_start - 1]):
+                bl, br = yppm.south_edge_jord8plus_0(v, dy, dm)
+
+            with horizontal(region[:, j_start]):
+                bl, br = yppm.south_edge_jord8plus_1(v, dy, dm)
+
+            with horizontal(region[:, j_start + 1]):
+                bl, br = yppm.south_edge_jord8plus_2(v, dm, al)
+                bl, br = yppm.pert_ppm_standard_constraint_fcn(v, bl, br)
+
+            with horizontal(region[:, j_end - 1]):
+                bl, br = yppm.north_edge_jord8plus_0(v, dm, al)
+                bl, br = yppm.pert_ppm_standard_constraint_fcn(v, bl, br)
+
+            with horizontal(region[:, j_end]):
+                bl, br = yppm.north_edge_jord8plus_1(v, dy, dm)
+
+            with horizontal(region[:, j_end + 1]):
+                bl, br = yppm.north_edge_jord8plus_2(v, dy, dm)
```

Comment:
In a later PR, maybe we can join these into a single function `blbr_edge_correction_jord8plus`, and maybe it could be called within `blbr_jord8`.